python/lsst/synpipe/FakeSourceLib.py

```diff
@@ -8,13 +8,35 @@
 import lsst.pex.config
 import lsst.afw.image
 import lsst.afw.cameraGeom
+import lsst.pipe.base as pipeBase
 
 
 """
 Helper functions for making fake sources
 """
 
 
+class SkyMapIdContainer(pipeBase.DataIdContainer):
+    """A version of lsst.pipe.base.DataIdContainer specialized for loading a skyMap.
+    Required because butler.subset does not support only tract
```

Comment:
I'm not the official reviewer, but I have two concerns about this class:

You are replacing use of CoaddDataIdContainer which performs the usual task of expanding incomplete data IDs. This class makes no attempt to replicate that behavior, yet all tasks should handle incomplete data IDs. I suggest you copy more of CoaddDataIdContainer if you must write your own.

It's a pity we got rid of chiSquared coadd dataset type, as that would be perfect for use with CoaddDataIdContainer since it has no filter. Non-filter-specific coadds are something we thought we would need, and I still think we'll need them.

Your code warns that `patch` will be ignored, yet `patch` is used to generate data references. I suggest you not warn for that. Or if the user really is not supposed to specify patch (which would be very unusual for a SkyMap) then you should remove the key instead of using it. If the `patch` key is not handled normally then please explain carefully in the doc string what's going on.

Reply:
Russel, I appreciate you glancing at this. I can assure you it works as intended, but it just looks a little weird because of the nature of the script that was previously written. The script in synpipe is using some of the lsst machinery but not in the way it was envisioned to be used. There is no effort available to rewrite large parts of synpipe in stack consistent ways, and really no need. These scripts will be used by a minority of users, and edited very infrequently. It was determined that the script have the best user interface (least surprises) it can, even if that comes at the sake of what is normally done in the stack.

It is a bit involved to explain how this fits in without getting into the whole context of the entire script. This code does not need to expand data ids, nor does it really even need much of a data reference. It does not need filter or patch. The script only wants a numeric tract number and that is what it gets with these changes.